### ISSUE TITLE
Run nightly tests as part of release testing

### ISSUE THREAD

**Issue by teskje** [MaterializeInc/materialize#16670]
The [current release playbook](https://www.notion.so/materialize/Release-playbook-3827577236a44cc39b885116554cf200) mandates selecting a commit that has passed CI on main, but it does not mandate that the commit has passed nightly tests (modulo known flakes). Neither is running the nightly tests part of the release testing that follows.

This leads to situations where we might discover critical bugs that make the nightlies fail only after the release has been cut and deployed to staging, which means we have to cut a new release with the fix and re-deploy. Or worse: We might not notice the nightlies failing if they don't block the release, which means we risk deploying avoidable bugs to production.

**Comment by teskje** [comment]
For example, release 0.36.2 would have been avoidable, because [the bug that it fixed](https://github.com/MaterializeInc/materialize/issues/16615) was detected by the nightly tests.

**Comment by umanwizard** [comment]
I think we can kick off the nightly in parallel to the other steps (deploying to staging, etc.), so that if the nightly is green we didn't waste several hours.

**Comment by benesch** [comment]
@philip-stoev is working on a replacement [Release qualification](https://buildkite.com/materialize/release-qualification) pipeline for integration into the canary process.

**Comment by benesch** [comment]
In fact I think it's already triggered automatically. It's just perhaps not part of the playbook to check the results yet?

**Comment by philip-stoev** [comment]
Nightly runs automatically on release tags. It ran on v0.36.1 and the regression was visible:

https://buildkite.com/materialize/nightlies/builds/1486


It was visible in v0.35.0 even:

https://buildkite.com/materialize/nightlies/builds/1483#01850827-acc0-47e1-be9f-b075bece7f5e

So, yes, the lesson here is to extend the playbook to check Nightly results.

**Comment by philip-stoev** [comment]
Actually, I would make the required additions to the playbook right now but I am not sure how to sequence the steps properly. Nightly takes ~2 hours, Release Qualification takes ~24 hours.

**Comment by chaas** [comment]
+1 I believe the upgrade tests only run on the nightly, and we want to be sure our release passes those tests.
The upgrade tests check for backwards-incompatible changes that would break customers' environments and prevent them from starting up.

**Comment by benesch** [comment]
Great, ok. I've added this under the "Verifying the canary release" section: https://www.notion.so/materialize/Release-playbook-3827577236a44cc39b885116554cf200#5a9340580a3c4e0e94f144a803b86b4b. Next year we should plan to choose the canary on Friday afternoon, qualify it over the weekend, and only roll out to staging canaries _after_ qualification succeeds, but I think it's a bit too soon for that. 

Thanks all! Reopen if any complaints with what I've added to the playbook.
